### ISSUE TITLE
And if i need to check ms(pong)? how can i check the pingInterval?

### ISSUE THREAD

**Issue by SamukaDEV** [doghappy/socket.io-client-csharp#36]

**Comment by doghappy** [comment]
Please upgrade to `1.0.3.9`

```cs
var client = new SocketIO("http://localhost:3000");
client.OnPong += () =>
{
    // to do
};
```

**Comment by SamukaDEV** [comment]
I didn't understand, where can I get the ms result? latency? an empty function was left !!

thanks for the update

**Comment by doghappy** [comment]
It hasn't been exposed yet. I think it's here.


https://github.com/doghappy/socket.io-client-csharp/blob/6d623399d43e25898d9489f766bab733ed87d479/SocketIOClient/SocketIO.cs#L195

**Comment by SamukaDEV** [comment]
sorry, but i dont know so mutch from CSharp, can u give me an example?
thanks a lot for the attention!

**Comment by doghappy** [comment]
OK, I have exposed an event, for example:

Please upgrade to `1.0.3.10`

```cs
var client = new SocketIO("http://localhost:3000");

// In the following two ways, you can choose what you like.

// 1. lambda
client.OnOpend += async args =>
{
    Console.WriteLine(args.PingInterval);
    Console.WriteLine(args.PingTimeout);
};

// 2. event agent method
client.OnOpend  += Clinet_OnOpend;

private void Client_OnOpend(OpenedArgs args)
{
    Console.WriteLine(args.PingInterval);
    Console.WriteLine(args.PingTimeout);
}
```

**Comment by SamukaDEV** [comment]
What if I need to check this periodically? for example every 5 seconds, in my view, your example will only be checked once when connected! right?

**Comment by doghappy** [comment]
When `socket.io` is connected, the server will send some data(OpendArgs) to the client to keep the heartbeat,  Therefore, you can only detect once after each successful connection.

If you want to detect every successful Ping, you can try this: https://github.com/doghappy/socket.io-client-csharp/issues/36#issuecomment-610212630 

If you want start a interval task, you can try this:

```cs
private void Client_OnOpend(OpenedArgs args)
{
    Task.Factory.StartNew(async () =>
    {
        while (true)
        {
            await Task.Delay(args.PingInterval);
            // to do
        }
    });
}
```

**Comment by SamukaDEV** [comment]
no, sorry, but i think u haven't understand what really i need, the latency, the socket server send a ms number in this pong event, and i need of this number, not of intercept the ping interval calls!

this args.PingInterval is a kind of socket client setting? in this case, a "constant" setting, do not?

look this image from Socket.io 
![SocketIO](https://user-images.githubusercontent.com/13206430/78790165-87403f00-7984-11ea-89c4-03859b2d01fb.PNG)

i need this ms number!

**Comment by doghappy** [comment]
Thansk :)
I have implemented this feature, Please upgrade to `1.0.3.11`

```cs
client.OnPong += span => Console.WriteLine("ping: " + span.TotalMilliseconds);
```

**Comment by SamukaDEV** [comment]
Thanks a lot!! done
